### ISSUE TITLE
CSSObj Really slow

### ISSUE THREAD

**Issue by PepijnSenders** [hellofresh/css-in-js-perf-tests#23]
@PhiLhoSoft CSSObj seems really slow, I think we're doing something wrong:

```
  aphrodite              x  1,548 ops/sec ±2.02% (80 runs sampled)
  aphrodite-no-important x  1,636 ops/sec ±3.19% (77 runs sampled)
  cssobj                 x    839 ops/sec ±2.66% (80 runs sampled)
  cxs                    x  3,038 ops/sec ±2.56% (76 runs sampled)
  cxs-optimized          x  2,249 ops/sec ±2.61% (79 runs sampled)
  fela                   x 26,995 ops/sec ±2.85% (81 runs sampled)
  free-style             x  3,157 ops/sec ±2.71% (72 runs sampled)
  glamor                 x  6,165 ops/sec ±3.16% (77 runs sampled)
  j2c                    x  3,640 ops/sec ±2.32% (75 runs sampled)
  jss                    x  3,449 ops/sec ±3.38% (75 runs sampled)
  jss-without-preset     x  5,401 ops/sec ±2.80% (78 runs sampled)
  styletron              x 22,047 ops/sec ±3.11% (79 runs sampled)
```

**Comment by PhiLhoSoft** [comment]
Perhaps, or perhaps it is just slow... 😄 
I have found nothing in the docs, so I will open an issue at their project to see if they can do something or advise a trick.

Notes:
I used [server-side rendering](https://github.com/cssobj/cssobj/wiki/Server-Side-Rendering), which, AFAIK, performs a bit slower than the client-side one, using gencss instead of cssom.
For this library, and for j2c, I have not found a reset / flush function, I don't know if that's a problem for perf.
Honestly, server-side perf is probably a minor point, taking 15 ms to generate CSS instead of 5 ms (made up figures...) doesn't seem a big deal... Perf on client is perhaps a bit more critical, but this project doesn't measure that (yet?).

**Comment by PepijnSenders** [comment]
I think you should use the flush functions, when they will build a big cache object on the global scope it will become very slow!

**Comment by PhiLhoSoft** [comment]
Of course, but as said, they don't seem to have one. But j2c doesn't seem to need one (at least it performs decently).
The CSSObj owner answered, he focuses more on the client side speed, particularly in the case of CSS updates (conditional styles, etc.).

**Comment by PepijnSenders** [comment]
Hehe okay, well for Server-Side rendering it's definitely not a good choice then 😛
